runtime/wasm/src/host.rs

```diff
@@ -90,17 +91,23 @@ where
     type Req = MappingRequest;
 
     fn spawn_mapping(
-        raw_module: &[u8],
+        raw_module: Vec<u8>,
         logger: Logger,
         subgraph_id: SubgraphDeploymentId,
         metrics: Arc<HostMetrics>,
-    ) -> Result<Sender<Self::Req>, Error> {
+    ) -> Result<Sender<Self::Req>, anyhow::Error> {
+        let timeout = std::env::var(TIMEOUT_ENV_VAR)
```

Comment:
Generally, fatal errors need not be deterministic.

Reply:
I'm not sure we are well-positioned to back that opinion. Who, for example, decides whether an indexer is eligible to claim inflation rewards? Some fatal errors are deterministic, and it would stand to reason that the error itself be a part of the total output of the act of indexing. The error, in the deterministic case, is something you can _attest_. As soon as we lose the ability to attest, we also lose the ability to verify claims, and with that goes the security of The Network.

Reply:
That's a valid point. At the moment we don't differentiate between deterministic and non-deterministic fatal errors, so failures cannot be attested to.